File: JPABookNook/src/main/java/com/skilldistillery/booknook/entities/Review.java
package com.skilldistillery.booknook.entities;

import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import org.hibernate.annotations.CreationTimestamp;

@Entity
public class Review {

	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	@ManyToOne
	@JoinColumn(name="book_id")
	private Book book;
	
	@ManyToOne
	@JoinColumn(name="user_id")
	private User user;
	
	private String content;
	
	private Integer rating;
	
	private boolean enabled;
	
	private String title;
	
	@Column(name="review_date")
	@CreationTimestamp
	private LocalDateTime reviewDate;


	
	
}

File: ngBookNook/src/app/models/review.ts
import { i18nMetaToJSDoc } from "@angular/compiler/src/render3/view/i18n/meta";
import { Title } from "@angular/platform-browser";
import { Book } from "./book";
import { User } from "./user";

export class Review {
  id: number;
  book: Book;
  user: User;
  content: string;
  rating: number;
  enabled: boolean;
  title: string;
  reviewDate: string;

  constructor(
  id: number = 0,
  book: Book = new Book(),
  user: User = new User(),
  content: string = '',
  rating: number = 0,
  enabled: boolean = true,
  title: string = '',
  reviewDate: string = ''
  ){
this.id = id;
this.book = book;
this.user = user;
this.content = content;
this.enabled = enabled;
this.rating = rating;
this.title = title;
this.reviewDate = reviewDate;
  }

}
